packages/client-utils/src/vstorage.js

```diff
@@ -1,30 +1,66 @@
+import { decodeBase64 } from '@endo/base64';
+import { encodeHex } from '@agoric/internal/src/hex.js';
+import { StreamCellShape } from '@agoric/internal/src/lib-chainStorage.js';
+import { mustMatch } from '@agoric/internal';
 import {
   QueryChildrenRequest,
   QueryChildrenResponse,
   QueryDataRequest,
   QueryDataResponse,
 } from '@agoric/cosmic-proto/agoric/vstorage/query.js';
-import { decodeBase64 } from '@endo/base64';
-import { encodeHex } from '@agoric/internal/src/hex.js';
 
 /**
+ * @import {AbciQueryResponse} from '@cosmjs/tendermint-rpc';
+ * @import {JsonSafe} from '@agoric/cosmic-proto';
+ * @import {StreamCell} from '@agoric/internal/src/lib-chainStorage.js';
  * @import {MinimalNetworkConfig} from './network-config.js';
  */
 
+/**
+ * @param {'data' | 'children'} kind
+ */
+const getCodecKey = kind => {
+  switch (kind) {
+    case 'children':
+      return '/agoric.vstorage.Query/Children';
+    case 'data':
+      return '/agoric.vstorage.Query/Data';
+    default:
+      throw Error(`invalid codec kind ${kind}`);
+  }
+};
```

Comment:
I'd rather see this as const data, to simplify the typing of functions that use it.  I won't block, though, if you'd prefer not to make this change.

```suggestion
const kindToRpc = {
  __proto__: null,
  children: '/agoric.vstorage.Query/Children',
  data: '/agoric.vstorage.Query/Data',
};
```

Reply:
Ah, that does help with the typings. I was trying to get a consistent error message but I'll restore the const.

Reply:
I ended up not using `keyof` to make the API docs more readable